src/ios/SimpleCameraPreview.m

```
@@ -259,16 +270,19 @@ - (void) capture:(CDVInvokedUrlCommand*)command {
         useFlash = false;
     self.photoSettings = [AVCapturePhotoSettings photoSettingsWithFormat:@{AVVideoCodecKey : AVVideoCodecTypeJPEG}];
     if (self.sessionManager != nil)
-        [self.sessionManager setFlashMode:useFlash? AVCaptureFlashModeOn: AVCaptureFlashModeOff photoSettings:self.photoSettings];
-
-    CDVPluginResult *pluginResult;
-    if (self.cameraRenderController != NULL) {
-        self.onPictureTakenHandlerId = command.callbackId;
-        [self.sessionManager.imageOutput capturePhotoWithSettings:self.photoSettings delegate:self];
-    } else {
-        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Camera not started"];
-        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
-    }
+    [self.sessionManager setFlashMode:useFlash? AVCaptureFlashModeOn: AVCaptureFlashModeOff photoSettings:self.photoSettings completion:^(BOOL success) {
+        if (success) {
```

Else part needed?

---

Yes, we still need to return an error when the camera isn’t started.

---

Agree with @dinitri.